Dockerfile

```diff
@@ -0,0 +1,67 @@
+FROM rocker/tidyverse:4.0.3
+
+# Update apt-get and install other libraries
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    curl \
+    libbz2-dev \
+    libglpk40 \
+    liblzma-dev \
+    libxt-dev \
+    python3-pip \
+    python3-dev
+
+# Install pyrefinebio v0.3.0
+RUN pip3 install pyrefinebio==0.3.0
+
+# R packages
+RUN install2.r --error --deps TRUE \
+    ape \
+    binr \
+    caret \
+    cluster \
+    corrplot \
+    cowplot \
+    data.table \
+    devtools \
+    doParallel \
+    dplyr \
+    e1071 \
+    fastICA \
+    flexclust \
+    fpc \
+    gdata \
+    ggplot2 \
+    ggupset \
+    glmnet \
+    gridExtra \
+    here \
+    Hmisc \
+    huge \
+    kernlab \
+    optparse \
+    parallel \
+    plyr \
+    ranger \
+    RColorBrewer \
+    reshape2 \
+    scales \
+    sdcMicro \
+    stringr \
+    styler \
+    viridis
+
+# R Bioconductor packages
+RUN Rscript -e "options(warn = 2); BiocManager::install(c( \
+  'limma', \
+  'quantro'), \
+  update = FALSE)"
+
+# Threading issue with preprocessCore::normalize.quantiles
+# https://support.bioconductor.org/p/122925/#124701
+# https://github.com/bmbolstad/preprocessCore/issues/1#issuecomment-326756305
+RUN Rscript -e "options(warn = 2); BiocManager::install( \
+    'preprocessCore', \
+    configure.args = '--disable-threading', \
+    update = FALSE)"
+
+RUN installGithub.r greenelab/TDM
```

Comment:
I'm not familiar with how `installGithub.r` works, but it may be useful to tie this to a particular commit or release with [`remotes::install_github()`](https://www.rdocumentation.org/packages/remotes/versions/2.3.0/topics/install_github). If `installGithub.r` freezes GH repos in some way or has additional advantages (e.g., better freezes CRAN dependencies, although it's not obvious to me that would be the case), you can feel free to ignore this comment.

Reply:
Updated to use `remotes::install_github` and reference the latest commit as of 2021-05-28.